### DISCUSSION TITLE
OmniSharp Visual Studio Code

### DISCUSSION THREAD

**Discussion by eoliverdev** [balta-io/forum#91] [Tira Dúvidas] [answered]
Olá Balta, Estou no tópico .NET de fundamento de C#. Quando abri o projeto no VS Code ele apareceu a seguinte mensagem:

"Omnisharp requires an install of the .NET SDK to provide language services when 'omnisharp.useModernNet' is enabled in Settings. Please install the latest .NET SDK and restart vscod. If you continue see this error after installing .NET and restarting vscode, you may need to log out and log back in or restart your system for changes to the PATH to take effect."

Com as opções "Open settings" e "Get the SDK".

Se eu desabilito as opções no "Open Settings" para de aparecer a mensagem mas não cria a pasta '.vscode'.

Mas instalar o SDK?! Eu já não instalei!!?? 

O que eu devo fazer?!

Obrigado!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @eoliverdev, eu tive este problema no Windows e se não me engano tive que instalar o .NET 5 também...
Todo caso, instala o .NET 5, 6 e também o 7 que saiu ontem 😂

Eles rodam lado a lado de boa!

**Comment by eoliverdev** [comment]
Instalei o 5 e nada, mas aí instalei o 6 e a mensagem não veio mais, e veio a mensagem igual a do seu vídeo para criação da pasta '.vscode' ... pasta criada, tudo ok. 

Mas ao dar um 'dotnet run' no projeto MeuApp ele me trouxe a msg a seguir que parece que ao invés de utilizar o 3.1 ele ta usando o 6 e o build falhou! O que eu faço?

Bem-vindo(a) ao .NET 6.0.
---------------------
Versão do SDK: 6.0.403

Telemetria
---------
As ferramentas do .NET coletam dados de uso para ajudar-nos a aprimorar sua experiência. Eles são coletados pela Microsoft e compartilhados com a comunidade. Você pode recusar a telemetria definindo a variável de ambiente DOTNET_CLI_TELEMETRY_OPTOUT como '1' ou 'true' usando seu shell favorito.    

Leia mais sobre a telemetria das Ferramentas da CLI do .NET: https://aka.ms/dotnet-cli-telemetry

----------------
Um certificado de desenvolvimento HTTPS ASP.NET Core foi instalado.
Para confiar no certificado, execute 'dotnet dev-certs https --trust' (somente Windows e macOS).
Saiba mais sobre HTTPS: https://aka.ms/dotnet-https
----------------
Escreva seu primeiro aplicativo: https://aka.ms/dotnet-hello-world
Descubra o que há de novo: https://aka.ms/dotnet-whats-new
Explore a documentação: https://aka.ms/dotnet-docs
Relate problemas e encontre a fonte no GitHub: https://github.com/dotnet/core
Use 'dotnet --help' para ver os comandos disponíveis ou visite: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
C:\dev\MeuApp\MeuApp.csproj : error NU1100: Não foi possível resolver 'Microsoft.AspNetCore.App.Ref (= 3.1.10)' para '.NETCoreApp,Version=v3.1'.
C:\dev\MeuApp\MeuApp.csproj : error NU1100: Não foi possível resolver 'Microsoft.NETCore.App.Host.win-x64 (= 3.1.31)' para '.NETCoreApp,Version=v3.1'.

Ocorreu uma falha no build. Corrija os erros de build e execute novamente.

**Reply by andrebaltieri** [reply]
@eoliverdev utiliza a flag `-f`

```
dotnet new console -o NomeDoApp -f net3.1
dotnet new console -o NomeDoApp -f net5.0
dotnet new console -o NomeDoApp -f net6.0
...
```
